### PR TITLE
Separate user-id translation component

### PR DESCRIPTION
**Pull request by petrKavulok**
translates a single user ID

https://user-images.githubusercontent.com/79644454/145082213-faf29d6f-0cac-4c02-bcff-eb16563a870f.mov

translates multiple user IDs

https://user-images.githubusercontent.com/79644454/145082221-7955f154-8c93-4db6-a0ac-fe4159cbea5d.mov

### CODE REVIEW

File: src/containers/UsernameTranslation.js
Comment (ateska) [inline src/containers/UsernameTranslation.js:49]: It is not a Row ... span or div ... check the DateTime object.

File: src/containers/UsernameTranslation.js
Comment (ateska) [inline src/containers/UsernameTranslation.js:59]: The "id" should be in the hint, so I can hover over the username and it reveals the id in the popup.

Also, consider using icon - in a same way as DateTime uses clock ... it will unify the look&feel.

**Review by ateska — CHANGES_REQUESTED** [review]

**Comment by ateska** [comment]
Also, use the local storage as a cache for translations ... it is quite static operation, so this will save a lot of server requests.

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:26]: @petrKavulok Can this be a time defined from the properties of the component? What I mean is, we should make it configurable

```
<UsernameTranslation cleanupTime={132456789} ... />
```

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:36]: @petrKavulok This will definitely not be a `TenantDetailContainer` but something like `ASABUsernameTranslation`

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:42]: @petrKavulok The same as mentioned above

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [reply src/containers/UsernameTranslation.js:26]: with some default e.g. 24h

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:46]: @petrKavulok `getUsernamesFromLS` would be better name

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:63]: @petrKavulok `setUsernamesToLS`

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

**Review by ateska — CHANGES_REQUESTED** [review]
Eventually the localstorage should be migrated to this: https://www.w3.org/TR/IndexedDB/

IndexedDB should be better technique than JSON in the LS/

**Comment by petrKavulok** [comment]
Case of a valid ID and an invalid IDs


https://user-images.githubusercontent.com/79644454/147088116-56381e31-9242-484b-bf2c-79aa49817467.mov

**Comment by ateska** [comment]
When resolve fails, don't display "N/A", keep there the original credential_id (be defensive).

**Comment by ateska** [comment]
... and the name of component is `<Credentials />` (it is to diplay credentials ... don't include name of the functionality in the title).

File: src/containers/UsernameTranslation.js
Comment (ateska) [inline src/containers/UsernameTranslation.js:19]: `Credentials` ... see my comment.

File: src/containers/UsernameTranslation.js
Comment (ateska) [inline src/containers/UsernameTranslation.js:50]: There is no `usernameIDs` or `UserIds` ... there are `credentials_id` and `username`s.

**Review by ateska — CHANGES_REQUESTED** [review]
Naming convention is not aligned with SeaCat Auth.

File: src/containers/UsernameTranslation.js
Comment (ateska) [inline src/containers/UsernameTranslation.js:133]: If not present, use `credentials_id` as username,

**Review by ateska — CHANGES_REQUESTED** [review]

File: src/containers/UsernameTranslation.js
Comment (petrKavulok) [reply src/containers/UsernameTranslation.js:50]: i thought `credentials_id` would stand for `'mongodb:ext:1234567890'` and `username` would be `'Ringo'`.
`usernameIDs` on line 50 expects to be an array of strings

File: src/index.js
Comment (petrKavulok) [inline src/index.js:16]: I haven't changed the containers name yet, as the latest changes would not be visible since it would be considered a new file, but it will be changed to `export { Credentials } from './containers/Credentials'` before potential merging

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:114]: @petrKavulok `saveUsernamesToLS`

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:111]: @petrKavulok Is it necessary to setup expiration with `cleanupTime` to empty credentials? In overall it is kinda strange to set up a cleanupTime in `getUsernamesFromLS`. From my point of view, `cleanupTime` is not necessary for this function. I would expect undefined expiration on empty credentials list or some predefined time, because why would you need to cleanup an empty list/local storage?

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:5]: I would expect broader information about this component in the `docs`, not in the code itself. What I mean is this:
```
At the moment works ONLY with apiPath='seacat_auth'
- userId (required, array) - user ids to be translated
- apiPath - default (and only working so far) 'seacat_auth'
- cleanupTime - default set to 24 hours
```
It should be a markdown file, so it should keep the markdown syntax then.

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:30]: @petrKavulok if the component is named as `Credentials`, it might be also good to rename `usernames` everywhere in the code to `credentials` (also the names of the consts)

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:15]: @petrKavulok You should update the example (there is no `id` if I am not mistaken)

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:52]: @petrKavulok You can use `Date.now()` directly here, instead of passing it in the variable, which is not used anywhere else (`currentTime`)

```
if (usernamesInLS.credentials == undefined || usernamesInLS.credentials.length === 0 || usernamesInLS.expiration <= Date.now())
```

File: src/containers/UsernameTranslation.js
Comment (Pe5h4) [inline src/containers/UsernameTranslation.js:47]: @petrKavulok This is kinda confusing me, why is here used `credentials_id` as a props name,
```
const matchUserIds = (credentials_id) =>
```
when it is clearly an array of user IDs? You are passing an array:

https://github.com/TeskaLabs/asab-webui/blob/afe4f174a64b803e78e6cdd7fdcbcedc9ce5eaef/src/containers/UsernameTranslation.js#L69

`credentials_id` should be used for single ID

I suggest to unify the naming, I guess this was some misunderstanding in this conversation: https://github.com/TeskaLabs/asab-webui/pull/218#discussion_r773895258

File: src/index.js
Comment (Pe5h4) [reply src/index.js:16]: I think it would not rewrite the whole file to the point that it would be seen as a new one and the changes would not be visible. Git is clever enough to avoid this.

**Review by Pe5h4 — CHANGES_REQUESTED** [review]

File: src/containers/UsernameTranslation.js
Comment (titkov49) [inline src/containers/UsernameTranslation.js:139]: You can simplify `global.localStorage.setItem(...)` to `localStorage.setItem(...)`. 
`localStorage` is already a global browser variable.

Comment (titkov49) [reply src/containers/UsernameTranslation.js:139]: Also I guess that there is no need to `try/catch` localStorage since it's not supported only by browsers which were released 15-20 years ago. Those browsers cannot even start our apps I guess

Screenshot 2021-12-22 at 18 30 58

Comment (titkov49) [reply src/containers/UsernameTranslation.js:139]: ^^^ Those two comments are appliable to all places where you use localStorage

**Review by titkov49 — CHANGES_REQUESTED** [review]
Otherwise good and approved from my side once those change requests are solved

File: src/containers/UsernameTranslation.js
Comment (ateska) [reply src/containers/UsernameTranslation.js:5]: Remark: We don't have "user ids", we have "credentials_ids" ... @byewokko

File: src/containers/UsernameTranslation.js
Comment (ateska) [reply src/containers/UsernameTranslation.js:111]: `cleanupTime` should be a globally set variable somewhere else than in the actual container ... it is not container specific, it is feature specific - so it should be handled by some kind of service.

File: src/containers/UsernameTranslation.js
Comment (ateska) [reply src/containers/UsernameTranslation.js]: * `credentials_id` == 'mongodb:ext:1234567890' ... correct
* `username` == "Ringo" ... correct
* `credentials_ids` should be the array of `credentials_id`

**Review by titkov49 — APPROVED** [review]

**Review by Pe5h4 — APPROVED** [review]
Looks fine to me

**Comment by petrKavulok** [comment]
@ateska would you please take a look at the changes?

**Review by ateska — APPROVED** [review]
Nice.

**Pull request merged**
